ext/fetch/20_headers.js

```diff
@@ -91,6 +92,32 @@ function fillHeaders(headers, object) {
 // deno-lint-ignore no-control-regex
 const ILLEGAL_VALUE_CHARS = new SafeRegExp(/[\x00\x0A\x0D]/);
 
+function checkForInvalidValueChars(value) {
+  for (let i = 0; i < value.length; i++) {
+    const c = StringPrototypeCharCodeAt(value, i);
+
+    if (c === 0x0a || c === 0x0d || c === 0x00) {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+// TODO(bartlomieju): make it LRU cache with 64/128 size limit.
+const cache = {};
```

Comment:
What about `Map` rather than object, to prevent `__proto__` and similar attacks? If you want to use a plain object, it should have a `null` prototype.

Reply:
Done, used `Map` instead.